lib/awful/layout/suit/tile.lua.in

```diff
@@ -156,26 +155,28 @@ local function tile_group(cls, wa, orientation, fact, group)
     size = math.min(size, available)
 
     local coord = wa[y]
-    local geom = {}
     local used_size = 0
     local unused = wa[height]
     for c = group.first,group.last do
         local i = c - group.first +1
-        geom[width] = size - cls[c].border_width * 2
-        geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
+        local geom = {}
+        geom[width] = size
+        geom[height] = math.floor(unused * fact[i] / total_fact)
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
```

Comment:
I've been thinking in circles about this issue, and it seems like this belongs in a separate PR. Because

 1. None of the other layouts try to adapt to size hints, they just leave the weird gaps as-is. Rather than try to fix just the tiled layout, I think it would be better if we could somehow ignore size hints in the layout code and instead handle them as another post-processing step. We would need a general size hints applying algorithm, and it would need to handle layouts like magnifier which have a mix of tiled/overlapping clients. Sounds tricky.
 2. Size hints conflict with useless gaps with a naive multi-pass algorithm. If you apply the size hints first, uniformly shrinking the clients to add gaps would most likely not honor the size hints, which defeats the point. If you apply the gaps first, size hints would most likely cause the gaps to become nonuniform and weird-looking. That's not the end of the word but ideally we would be able to take into account the gaps naturally created by the size hints when adding useless gaps.

Honestly, trying to nicely tile the clients while also honoring the size hints seems like an impossible problem (or at least an NP-complete one). You're always going to end up with weird gaps or clients that aren't the right size.